sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -246,6 +248,32 @@ class Analyzer(
       CleanupAliases)
   )
 
+  /**
+   * Replace interval with startEnd + interval and substitute child plan with DateTimeOverlaps.
+   */
+  object ResolveDateTimeOverlaps extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
+      case p: LogicalPlan => p.transformAllExpressions {
+        case u @ UnresolvedDateTimeOverlaps(leftStart, leftEnd, rightStart, rightEnd) =>
+          withPosition(u) {
+            u.checkInputDataTypes() match {
```

Comment:
We cannot use `ExpectsInputTypes` instead?

Reply:
no, we cannot. It‘s `RuntimeReplaceable`.

Reply:
no, we cannot. It‘s `RuntimeReplaceable`.